lib/new_relic/agent/new_relic_service.rb

```diff
@@ -22,8 +22,15 @@ class NewRelicService
       CONNECTION_ERRORS = [Timeout::Error, EOFError, SystemCallError, SocketError].freeze
 
       # Don't perform compression on the payload unless its uncompressed size is
-      # greater than this number of bytes
-      MIN_BYTE_SIZE_TO_COMPRESS = 64 * 1024
+      # greater than or equal to this number of bytes. In testing with
+      # Ruby 2.2 - 3.1, we determined an absolute minimum value for ASCII to be
+      # 535 bytes to obtain at least a 10% savings in size. It is recommended
+      # that this value be kept above that 535 number. It is also important to
+      # consider the CPU cost involved with performing compression and to find
+      # a balance between CPU cycles spent and bandwidth saved. A good
+      # reasonable default here is 2048 bytes, which is a tried and true Apache
+      # Tomcat default (as of v8.5.78)
```

Comment:
I really appreciate the detailed comment explaining the WHY of the new minimum value 👍